app/grandchallenge/hanging_protocols/models.py

```diff
@@ -0,0 +1,102 @@
+from django.contrib.auth import get_user_model
+from django.db import models
+from guardian.shortcuts import assign_perm
+
+from grandchallenge.core.models import TitleSlugDescriptionModel, UUIDModel
+from grandchallenge.core.validators import JSONValidator
+
+
+class ImagePort(models.TextChoices):
+    MAIN = "M", "Main"
+    SECONDARY = "S", "Secondary"
+    TERTIARY = "TERTIARY", "Tertiary"
+    QUATERNARY = "QUATERNARY", "Quaternary"
+    QUINARY = "QUINARY", "Quinary"
+    SENARY = "SENARY", "Senary"
+    SEPTENARY = "SEPTENARY", "Septenary"
+    OCTONARY = "OCTONARY", "Octonary"
+    NONARY = "NONARY", "Nonary"
+    DENARY = "DENARY", "Denary"
+
+
+HANGING_PROTOCOL_SCHEMA = {
+    "definitions": {},
+    "$schema": "http://json-schema.org/draft-06/schema#",
+    "title": "The Hanging Protocol Schema",
+    "type": "array",
+    "contains": {
+        "type": "object",
+        "properties": {
+            "viewport_name": {
+                "type": "string",
+                "enum": [port.lower() for port in ImagePort.labels],
+                "pattern": "^main$",
+            },
+        },
+    },

```

Comment:
What does this do exactly? The `items` part below already validates the schema correctly so this does not add any extra functionality, right? Also, how does the combination of `pattern` and `enum` fields work?

Reply:
This validates that at least one of the items in the array has `main` as the `viewport_name`.  The items part below only validates that the `viewport_name`  is one of the strings specifed in the `enum` but not that the `main` viewport is required. Now that I look at it, the `enum` might not be necessary in the `contains` part here, because `pattern` is already specific enough.

Reply:
Ah right, makes sense!